force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -101,17 +112,18 @@ public with sharing class ClassesObjectsMethods {
      * @return The number of vowels in the string.
      */
     public static Integer countVowels(String str) {
-        // Initialize the result Integer
-
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
+        Integer numberOfVowels = 0;
+        List<String> vowelList = new List<String>{'a' , 'e' , 'i' , 'o' , 'u'};
+        List<String> listOfStr = str.split(' ');
+
+        for(String s: listOfStr) {
+            for(Integer i = 0; i < vowelList.size(); i++) {
+                if(s.countMatches(vowelList[i]) > 0) {
+                    numberOfVowels += s.countMatches(vowelList[i]);
+                }
+            }  
```

Comment:
Very nice use of the `countMatches()` method! But when we find ourselves creating a loop inside a loop, this is often a sign that there might be an easier way to meet our requirement. While there are many ways to do this, one way might look like this. (Notice here that we're using `str` and not `listOfStr`):
```suggestion
for (String vowel : vowelList) {
    numberOfVowels += str.toLowerCase().countMatches(vowel);
}
```
Notice too that we can remove the if statement since countMatches() may simply return 0. Let me know if you have questions about this!